backends/p4fmt/p4fmt.cpp

```diff
@@ -1,15 +1,46 @@
 #include "backends/p4fmt/p4fmt.h"
 
+#include "backends/p4fmt/attach.h"
+#include "backends/p4fmt/p4formatter.h"
 #include "frontends/common/parseInput.h"
 #include "frontends/common/parser_options.h"
+#include "frontends/parsers/parserDriver.h"
 #include "ir/ir.h"
 #include "lib/compile_context.h"
 #include "lib/error.h"
+#include "lib/source_file.h"
 #include "options.h"
-#include "p4formatter.h"
 
 namespace P4::P4Fmt {
 
+std::optional<std::pair<const IR::P4Program *, const Util::InputSources *>> parseProgram(
+    const ParserOptions &options) {
+    auto *file = fopen(options.file.c_str(), "r");
```

Comment:
what is the purpose of `fopen` here? Just for `"no such file" diagnostics as `file` is unused below? What does `parseProgramSources` reports in case no input file?

Reply:
Switched to `std::filesystem::exists` , if that helps?

Reply:
Do you really need this? What does parseProgramSources repors in case non-existent input file?

Reply:
I believe this is necessary as `parseProgramSources` doesn't report anything where there's no input file it just fails to parse and returns null.